src/gazebo_lidar_plugin.cpp

```diff
@@ -108,6 +109,84 @@ void RayPlugin::Load(sensors::SensorPtr _parent, sdf::ElementPtr _sdf)
   string topicName = "~/" + names_splitted[0] + "/link/lidar";
 
   lidar_pub_ = node_handle_->Advertise<sensor_msgs::msgs::Range>(topicName, 10);
+
+  // Precaution: remove empty strings
+  names_splitted.erase(std::remove_if(begin(names_splitted), end(names_splitted),
+                              [](const string& name)
+                              { return name.size() == 0; }), end(names_splitted));
+
+  std::string rootModelName = names_splitted.front(); // The first element is the name of the root model
+
+  // Get the pointer to the root model
+#if GAZEBO_MAJOR_VERSION >= 9
+  physics::ModelPtr rootModel = this->world->ModelByName(rootModelName);
+#else
+  physics::ModelPtr rootModel = this->world->GetModel(rootModelName);
+#endif
+
+  // Get the `base_link` rotation WRT world
+  physics::LinkPtr baseLink = nullptr;
+  std::vector<physics::LinkPtr> linkList = rootModel->GetLinks(); // Get list of all links in the root model
+  for(auto link : linkList) {
+    std::string linkName = link->GetName();
+    if(linkName.find("::base_link") != std::string::npos) {
+      baseLink = rootModel->GetLink(linkName); // Get the pointer to the `base_link`
+      break;
+    }
+  }
+  if (!baseLink)
+    gzthrow("RayPlugin requires the `base_link` element to be defined");
+
+  // This is the rotation of the 'base_link` WRT world
+#if GAZEBO_MAJOR_VERSION >= 9
+  ignition::math::Quaterniond q_wb = baseLink->WorldPose().Rot();
+#else
+  ignition::math::Quaterniond q_wb = ignitionFromGazeboMath(baseLink->GetWorldPose()).Rot();
+#endif
+
+  // Get the parent sensor link rotation WRT world
```

Comment:
I think this comment is not correct. You seem to be looking up the link name of the parent here.

I also don't see why the world transform would be relevant here, given that everything is wrt body.

Reply:
> given that everything is wrt body.

Im not sure. In fact it was the main reason why i did check if the parent link is the base link. IIRC every model and every link inside it defines its pose wrt world, not wrt its parent model or link. Thats why rotation wrt base link is computed here by using rotation wrt world. It would be convenient to have a Gazebo function that takes into account the 'parent-child' relations of models and links can return the pose for any of two given models or links in the hierarchy. However i didnt find it so i used global frame as reference here.